2.0/m_autooper.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* $ModAuthor: Sebastian Nielsen */
+/* $ModAuthorMail: sebastian@sebbe.eu */
+/* $ModDesc: Adds a opertype option to connect blocks so operators can get autologged in via IP */
+/* $ModDepends: core 2.0 */
+/* $ModConfig: Within connect block: opertype="OPER TYPE HERE" */
+
+
+#include "inspircd.h"
+
+class ModuleAutoOper : public Module
+{
+public:
+        void init()
+        {
+                Implementation eventlist[] = { I_OnUserConnect };
```

Comment:
If you're only hooking onto one event then you can just pass it to Attach rather than using an implementation list.